<a href="https://colab.research.google.com/github/OrastaRakhmatullayeva/ML-projects/blob/main/vector_and_matrix_practise.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
v=np.array([10,20,30])
print(v.shape)
print(v.ndim)
print(len(v))

(3,)
1
3


In [3]:
a=np.array([[[1,2],
              [3,4],
              [9,8]]])

print("the shape : ",a.shape)
print("the size: ",a.size)
print("the dimension : ",a.ndim)

the shape :  (1, 3, 2)
the size:  6
the dimension :  3


In [6]:
print(np.zeros(4))
print(np.ones(3))
print(np.eye(2))

[0. 0. 0. 0.]
[1. 1. 1.]
[[1. 0.]
 [0. 1.]]


In [7]:
a = np.array([2, 5, 8])
b=np.array([1,3,7])

print("the sum: ",a+b)
print("the substraction: ",a-b)

the sum:  [ 3  8 15]
the substraction:  [1 2 1]


In [8]:
narx = np.array([120, 240, 360])
discount=narx*0.75

print(discount)

[ 90. 180. 270.]


In [ ]:
np.array([1, 2]) + np.array([1, 2, 3])
#it returns error bcs it doesnot match the number of columns and rows

In [10]:
v=np.array([6,8])
print(np.sqrt(np.sum(v**2)))
print(np.linalg.norm(v))


10.0
10.0


In [12]:
a=np.array([1,0,2])
b=np.array([3,4,1])
print(a[0]*b[0]+a[1]*b[1]+a[2]*b[2])
print(a@b)

5
5


In [13]:
A = np.array([[10, 20, 30],
              [40, 50, 60],
              [70, 80, 90]])
print(A[1,2])
print(A[0])
print(A[:,1])
print(A[1:,:2])

60
[10 20 30]
[20 50 80]
[[40 50]
 [70 80]]


In [19]:
print("the elements of last column:",A[:,2])

print("all elements of 2-row:",A[1])

the elements of last column: [30 60 90]
all elements of 2-row: [40 50 60]


In [22]:
A = np.array([[1, 2], [3, 4]])
B = np.array([[2, 0], [1, 2]])

print("the scalar multiplacation: ",A*B)
print("dot product: ",A@B)

#scalar ko'paytamda: element by element ko'paytiriladi
# dot productda esa ko'paytmalar yig'indisi hisoblanadi

the scalar multiplacation:  [[2 0]
 [3 8]]
dot product:  [[ 4  4]
 [10  8]]


(2 x 3) @ (3 x 4)  it works

(3 x 2) @ (3 x 4)  it doesnot work

(4 x 2) @ (2 x 5)  it works

(2 x 2) @ (2 x 2)  it works



In [25]:
X = np.array([[25, 3.2],
              [34, 5.1],
              [41, 6.8]])   # yosh, daromad
w = np.array([1.5, 20])   #1.5 — yosh 1 birlikka (1 yoshga) oshsa, natija 1.5 birlikka ortadi
                          #20 — daromad 1 birlikka oshsa, natija 20 birlikka ortadi
b = 10

y=X@w+b
print(y)

print("For the first object: ",X[0]@w+b)

[111.5 163.  207.5]
111.5


In [26]:
A = np.array([[1, 2, 3],
              [4, 5, 6]])
print(A+100)
v=np.array([10,20,30])
print(A+v)

[[101 102 103]
 [104 105 106]]
[[11 22 33]
 [14 25 36]]


In [37]:
m1 = np.array([5, 0, 3, 1])
m2 = np.array([4, 0, 2, 1])
m3 = np.array([0, 5, 1, 4])

#the relation between m1 and m2
cos=((m1@m2)/(np.sqrt(np.sum(m1**2))*np.sqrt(np.sum(m2**2))))
print(cos)

#the relation between m1 and m3
cos1=((m1@m3)/(np.sqrt(np.sum(m1**2))*np.sqrt(np.sum(m3**2))))
print(cos1)

#conclusion m1 and m2 is too close and similar bcs near to 1, but m1 and mm3 is not similar

0.9959100033104787
0.18257418583505539


In [40]:
X=np.array([[30,2],
            [35,3],
            [40,3],
            [45,3],
            [50,3]])
w=np.array([15,30])
b=50
narx=X@w+b
print(narx)

print("the index of the highest price home:",np.argmax(narx))

[560 665 740 815 890]
the index of the highest price home: 4


In [42]:
X=np.array([[[30,2,1],
            [35,3,2],
            [40,3,1],
            [45,3,4],
            [50,3,3]]])
w=np.array([15,30,2])
b=50
narx=X@w+b
print(narx)

print("the index of the highest price home:",np.argmax(narx))

[[562 669 742 823 896]]
the index of the highest price home: 4


uyning maydoni 2 barobarga oshirsak, narx 2 barobarga ortmaydi, chunki dot product hisob kitobiga bog;liq bo'ladi. misol uchun:


In [43]:
X=np.array([[60,2],
            [70,3],
            [80,3],
            [90,3],
            [100,3]])
w=np.array([15,30])
b=50
narx=X@w+b
print(narx)

print("the index of the highest price home:",np.argmax(narx))

[1010 1190 1340 1490 1640]
the index of the highest price home: 4


[560 665 740 815 890] bu avvalgi narxlar, hozir esa [1010 1190 1340 1490 1640]

xulosa maydon 2 marta ortgani bilan natija 2 barobarga ortadi degan statement false qiymat qaytaradi


